src/response.ts

```diff
@@ -0,0 +1,94 @@
+import { Entity } from '@siren-js/core';
+
+/**
+ * [Decorates](https://en.wikipedia.org/wiki/Decorator_pattern) a
+ * [`Response`](https://developer.mozilla.org/en-US/docs/Web/API/Response)
+ * object with the `siren()` method for parsing JSON
+ * [Siren](https://github.com/kevinswiber/siren) (`application/vnd.siren+json`).
+ */
+export default class ClientResponse implements Response {
```

Comment:
`isSiren()` helper method would be useful.

Reply:
Seems odd to have that without `isJson()`, `isText()`, etc. helpers, which aren't part of the `Response` interface. What would `isSiren()` do exactly? Keep in mind, that the body is a stream, so once it's read, it cannot be read again.

Reply:
It would check that the `Content-Type` header is Siren (maybe `isContentTypeSiren()`). That will probably be a common check.

Reply:
I don't believe checking the `Content-Type` header is sufficient, per [Section 3.1.1.5 of RFC 7231](https://tools.ietf.org/html/rfc7231#section-3.1.1.5):

> A sender that generates a message containing a payload body SHOULD
   generate a Content-Type header field in that message unless the
   intended media type of the enclosed representation is unknown to the
   sender.  If a Content-Type header field is not present, the recipient
   MAY either assume a media type of "application/octet-stream"
   ([[RFC2046], Section 4.5.1](https://tools.ietf.org/html/rfc2046#section-4.5.1)) or examine the data to determine its type.

The last piece ("examine the data to determine its type") is where it gets tricky with the body stream. Since there are no other `isMediaType()` helpers, I'm not sure there's much value here. If it's not Siren, users will have to check the `Content-Type` using the typical `headers.get('Content-Type')` anyway. OTOH, it might be useful to have a constant for the Siren media type, so the check would be less error-prone. Something like this, maybe?

```js
if (response.headers.get('Content-Type') === MediaType.Siren) {
 const entity = await response.siren();
}
```